src/internal/metrics/metrics.go

```diff
@@ -261,18 +261,6 @@ func (r *Reporter) internalMetrics(metrics *Metrics) {
 		}
 		metrics.Pipelines = int64(len(pipelineInfos)) // Number of pipelines
 		for _, pi := range pipelineInfos {
-			if pi.Details.ParallelismSpec != nil {
```

Comment:
If not here in the loop, where do we update the metrics for max/min parallelism?

Reply:
It's actually lower down in the file, I think this is only here because of an incorrect merge

Reply:
Are we ever going to be able to get the pipeline details with auth enabled? They're populated by reading the spec commit from PFS, but this code runs with no credentials - I don't think we have an internal code path that can read spec commits without credentials ATM.

Reply:
Ah, `GetPipelineDetails` attempts to use the auth token from the pipeline info that gets passed in, but `listPipelineInfo` strips it. I'm confused right now - I don't see any code path that tries to only return the subset of pipelines which the caller is authorized to access, so it seems like if auth is enabled and you don't have access to all repos, `pachctl list pipeline` will error.

Reply:
Now I'm not sure how this is a problem - it seems like exactly this behavior is tested in the auth test `TestCreateAndUpdatePipeline`, enabled by #6432, so I don't think we need auth?

Reply:
Ok, to summarize the problem: we enforce authorization on internal gRPCs, which meant that even the ListRepo here would fail without credentials (even though it doesn't actually need permissions). The ListPipeline additionally needs to pass an auth check to read the spec repo (which again will work with any non-empty credentials). 

To solve these problems, I added more convenient wrappers for the non-gRPC versions of relevant functions (since I think we're still trying to avoid unnecessary gRPCs?) and made metrics try to request a robot token with no permissions.